lib/Frontend/ParseableInterfaceSupport.cpp

```diff
@@ -178,11 +178,28 @@ class InheritedProtocolCollector {
   SmallVector<const ProtocolType *, 8> ConditionalConformanceProtocols;
 
   /// Helper to extract the `@available` attributes on a decl.
-  AvailableAttrList getAvailabilityAttrs(const Decl *D) {
-    AvailableAttrList result;
-    auto attrRange = D->getAttrs().getAttributes<AvailableAttr>();
-    result.insert(result.begin(), attrRange.begin(), attrRange.end());
-    return result;
+  static AvailableAttrList
+  getAvailabilityAttrs(const Decl *D, Optional<AvailableAttrList> &cache) {
+    if (cache.hasValue())
+      return cache.getValue();
+
+    cache.emplace();
+    while (D) {
+      for (auto *nextAttr : D->getAttrs().getAttributes<AvailableAttr>()) {
+        // FIXME: This is just approximating the effects of nested availability
+        // attributes for the same platform; formally they'd need to be merged.
+        bool alreadyHasMoreSpecificAttrForThisPlatform =
+            llvm::any_of(*cache, [nextAttr](const AvailableAttr *existingAttr) {
+          return existingAttr->Platform == nextAttr->Platform;
```

Comment:
Does this rely on Sema diagnosing when an inner thing is available on an earlier platform than the outer thing?

Reply:
Yeah, it does. That'd be a weird thing to do, though, since it means you can use the type with one availability guard but you can't name it without a stricter one.

Reply:
Yeah, it doesn't make sense, but I wanted to make sure we were explicitly not verifying that the existing one is earlier than the inner one here. 👍